### ISSUE TITLE
`skipToNext` while track is repeating does not work on iOS

### ISSUE THREAD

**Issue by mmmoussa** [DoubleSymmetry/react-native-track-player#1433]
**Describe the Bug**
On iOS, calling `TrackPlayer.skipToNext` with `track` repeat mode causes the trackplayer to get stuck. Trying to reset the trackplayer at this point causes a crash. Behaviour is fine on Android.

**Steps To Reproduce**
1. On iOS, create a Queue with two songs
2. Set the repeat mode to `RepeatMode.Track`
3. Start playing the queue
4. Call `TrackPlayer.skipToNext`
5. Call `TrackPlayer.reset`

**Code To Reproduce**
```
await  TrackPlayer.add([{url: 'sampleUrl', title: 'Title 1', artist: 'Artist 1'}, {url: 'sampleUrl2', title: 'Title 2', artist: 'Artist 2'}]);
await TrackPlayer.setRepeatMode(RepeatMode.Track);
await TrackPlayer.play();
// First track now playing correctly on repeat
...
await TrackPlayer.skipToNext();
// TrackPlayer no longer playing
...
await TrackPlayer.reset()
// App has now crashed
```

**Environment Info:**
`react-native`: 0.67.3
`react-native-track-player`: 2.2.0-rc2
Tested on both simulator and real device running iOS 15. Also tested on Android.

**How I can Help**
Did some preliminary investigation. I'm not sure why the trackplayer gets stuck when trying to skip to next, but when calling `reset` afterwards, the swift `handleAudioPlayerPlaybackEnded` method is where the crash happens due to the player not having a `currentIndex` value anymore. I am very unfamiliar with Swift but will try to get a better grasp of the issue and share that at least if not put up a PR.

**Comment by mmmoussa** [comment]
I realized that the crash on reset was an entirely separate problem and fixed it with #1434, but I'm not sure what the cause of skipping not working is.

**Comment by mmmoussa** [comment]
Debugged the problem further and now closing in favour of https://github.com/DoubleSymmetry/SwiftAudioEx/issues/12
